code/game/mecha/mech_fabricator.dm

```diff
@@ -281,151 +320,103 @@
+
+	if(href_list["main_menu_add_all"])
+		selected_category = href_list["main_menu_add_all"]
+		for(var/I in files.known_designs)
+			var/datum/design/D = files.known_designs[I]
+			if(D.build_type & fabricator_type && selected_category in D.category)
+				add_to_queue(D)
+		selected_category = null
+
 	if(href_list["remove_from_queue"])
-		remove_from_queue(afilter.getNum("remove_from_queue"))
-		return update_queue_on_page()
-	if(href_list["partset_to_queue"])
-		add_part_set_to_queue(afilter.get("partset_to_queue"))
-		return update_queue_on_page()
+		var/index = text2num(href_list["remove_from_queue"])
+		remove_from_queue(index)
+
 	if(href_list["process_queue"])
-		spawn(0)
-			if(processing_queue || being_built)
-				return FALSE
-			processing_queue = 1
-			process_queue()
-			processing_queue = 0
-	if(href_list["clear_temp"])
-		temp = null
-	if(href_list["screen"])
-		screen = href_list["screen"]
-	if(href_list["queue_move"] && href_list["index"])
-		var/index = afilter.getNum("index")
-		var/new_index = index + afilter.getNum("queue_move")
-		if(isnum(index) && isnum(new_index))
-			if(IsInRange(new_index,1,queue.len))
-				queue.Swap(index,new_index)
-		return update_queue_on_page()
+		if(processing_queue || being_built)
+			return FALSE
+		processing_queue = TRUE
+		process_queue()
+		processing_queue = FALSE
+
 	if(href_list["clear_queue"])
 		queue = list()
-		return update_queue_on_page()
+
+	if(href_list["queue_move"] && href_list["index"])
+		var/new_index = text2num(href_list["queue_move"])
+		var/index = text2num(href_list["index"])
+		if(isnum(index) && isnum(new_index))
+			if(IsInRange(new_index, 1, queue.len))
+				queue.Swap(index, new_index)
+
 	if(href_list["sync"])
 		sync()
-	if(href_list["part_desc"])
-		var/T = afilter.getStr("part_desc")
-		for(var/v in files.known_designs)
-			var/datum/design/D = files.known_designs[v]
-			if(D.build_type & fabricator_type)
-				if(D.id == T)
-					var/obj/part = D.build_path
-					temp = {"<h1>[initial(part.name)] description:</h1>
-								[initial(part.desc)]<br>
-								<a href='?src=[UID()];clear_temp=1'>Return</a>
-								"}
-					break
 
-	if(href_list["remove_mat"] && href_list["material"])
+	if(href_list["material_storage"])
+		screen = MECHFAB_MATERIALS_MENU
+
+	if(href_list["remove_mat"])
 		GET_COMPONENT(materials, /datum/component/material_container)
-		materials.retrieve_sheets(text2num(href_list["remove_mat"]), href_list["material"])
+		var/desired_num_sheets
+		if(href_list["remove_mat_amount"] == "custom")
+			desired_num_sheets = input("How many sheets would you like to eject from the machine?", "How much?", 1) as null|num
+			desired_num_sheets = max(0, desired_num_sheets) // If you input too high of a number, the mineral datum will take care of it either way
+			if(!desired_num_sheets)
+				return
+			desired_num_sheets = round(desired_num_sheets) // No partial-sheet goofery
+		else
+			desired_num_sheets = text2num(href_list["remove_mat_amount"])

```

Comment:
href_list is user-controlled ("attacker-controlled" if you wish), you still need to sanitize it the same way you do for custom amount (just without the `input()`)